### PR TITLE
[WIP] Fix build without dealii (clang-format not applied yet)

### LINKED ISSUE #261: Bug: current version won't build if deal.II is not enabled

### PR DESCRIPTION
**Pull request by dalg24**
fix #261

### CODE REVIEW

**Comment by dalg24** [comment]
We might need a runtime api call that tells whether Cap was compiled with deal.II or not to emulate the `#ifdef WITH_DEAL_II` in Python.

File: cpp/example/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,10 @@
 include(${CMAKE_SOURCE_DIR}/cmake/UnitTesting.cmake)
 
+if(ENABLE_DEAL_II)
```
Comment (Rombur) [inline cpp/example/CMakeLists.txt:3]: Can you indent the lines below. It makes it easier to see what's in the if.

**Review by Rombur — APPROVED** [review]

File: cpp/example/CMakeLists.txt (hunk 1 above)
Comment (dalg24) [reply cpp/example/CMakeLists.txt:3]: I expect clang-format will do that.

Comment (dalg24) [reply cpp/example/CMakeLists.txt:3]: Oh right this is CMake...  I will do that.

**Comment by masterleinad** [comment]
Any reason not to merge this after formatting?

**Comment by Rombur** [comment]
I think that would be fine but I don't know if @dalg24 ever tried to compile without deal.II (I assume that's why it's `WIP`)

**Comment by dalg24** [comment]
We need a way to find out whether Cap was built with deal.II or nor from the Python side and then we need to conditionally enable the tests with the supercapacitor.  I remember I was looking at how NumPy records whether it was built with BLAS.

**Comment by masterleinad** [comment]
Isn't that a separate issue besides just being able to compile without `deal.II`?

**Comment by dalg24** [comment]
> Isn't that a separate issue besides just being able to compile without `deal.II`?

Yes and no.  It will compile but the tests won't pass.

**Pull request closed without merging**